source/extensions/filters/http/jwt_authn/authenticator.cc

```diff
@@ -270,6 +288,32 @@ void AuthenticatorImpl::verifyKey() {
   handleGoodJwt(/*cache_hit=*/false);
 }
 
+void AuthenticatorImpl::addJWTClaimToHeader(const std::string& claim_name,
+                                            const std::string& header_name) {
+  StructUtils payload_getter(jwt_->payload_pb_);
+  const ::Envoy::ProtobufWkt::Value* claim_value;
+  const auto status = payload_getter.GetValue(claim_name, claim_value);
+  std::string str_claim_value;
+  if (status == StructUtils::OK) {
+    if (claim_value->kind_case() == Envoy::ProtobufWkt::Value::kStringValue) {
```

Comment:
This whole block should be a separate utility function with a complete set of tests for all behaviors. It's got basic coverage below, but I'm not convinced all the various branching cases and state space is covered. This is security critical, so I think this is worth doing.

Reply:
What do you think is not covered in the tests ? imo we have added all the possible tests

Reply:
Updated the testing token with an array type value (https://github.com/envoyproxy/envoy/pull/23444/files#diff-ccb8c0f9dfe4011484c9d7ca83d56abbe8b60c7b1b8dcd8efbc3debe0c6c5a1cR237) and added one more test on this.